Makefile

```diff
@@ -0,0 +1,29 @@
+BIN_DIR := hack
+MDBOOK_BIN := $(BIN_DIR)/mdbook
+MDBOOK_VERSION := v0.4.3
+MDBOOK_RELEASE_URL := https://github.com/rust-lang/mdBook/releases/download/$(MDBOOK_VERSION)/mdbook-$(MDBOOK_VERSION)-x86_64-unknown-linux-gnu.tar.gz
+SOURCE_PATH := docs/user-guide
+
+## ----------------------
+## Documentation tooling
+## ----------------------
+
+.PHONY: bin
+bin:
+	curl -L $(MDBOOK_RELEASE_URL) | tar xvz -C $(BIN_DIR)
+
+.PHONY: serve
+serve: 
```

Comment:
These targets should depend on the `bin` target so the user does not need to run that step by itself. If that target is renamed to `$(MDBOOK_BIN)` then `make` will automatically skip it when it is not needed.

Reply:
Good point. Updated it. Thanks.